### PR TITLE
Show compiler version info

### PR DESCRIPTION
**Pull request by puqeko**
Assumes #46. See comments

### CODE REVIEW

File: db/languages.yml [hunk 1]
```diff
@@ -7,24 +7,30 @@ c++:
...
   :processes: 1
   :variants:
-    c++03:
+    c++03:  # depreciated
       :name: "C++03"
       :compiler: /usr/bin/g++
-      :compiler_command: "%{compiler} -std=gnu++03 -O2 -o %{output} %{source} -lm"
-#      upgrade: c++11 # submissions will be upgraded if possible, if support is removed, it will be force-upgraded
-#      note: probably will never happen for this particular example... (or at least not for the next 10 years)
+      :compiler_command: "%{compiler} --version 1>&2 && %{compiler} -std=gnu++03 -O2 -o %{output} %{source} -lm"
```
Comment (tom93) [inline db/languages.yml]: How about we only show the first line of the version (without the copyright)?
```
"%{compiler} --version | head -n 1 >&2 && ..."
```

File: db/languages.yml [hunk 2]
```diff
@@ -52,7 +63,7 @@ java:
     java6:
       :name: Java 1.6
       :compiler: /usr/bin/javac;/usr/bin/jar
-      :compiler_command: '%{compiler[0]} -O -source 1.6 -J-Xms16m -J-Xmx256m %{source} && %{compiler[1]} cfe %{output} Main -C %{source_dir} Main.class'
+      :compiler_command: '%{compiler[0]} -version 1>&2 && %{compiler[1]} --version 1>&2 && %{compiler[0]} -O -source 1.6 -J-Xms16m -J-Xmx256m %{source} && %{compiler[1]} cfe %{output} Main -C %{source_dir} Main.class'
```
Comment (tom93) [inline db/languages.yml]: I don't think anybody cares about the version of `jar`.

File: db/languages.yml (hunk 1 above)
Comment (puqeko) [reply db/languages.yml]: Good idea

File: db/languages.yml (hunk 2 above)
Comment (puqeko) [reply db/languages.yml]: Fair enough

**Comment by tom93** [comment]
The submission page shows the compilation command. I think ideally it should exclude the part that displays the version, but it's not important and can be changed later.

**Comment by puqeko** [comment]
> The submission page shows the compilation command. I think ideally it should exclude the part that displays the version, but it's not important and can be changed later.

Prehaps there needs to be another field in the .yml for a version command which can be run on startup. For now, this should do

**Comment by puqeko** [comment]
2020-01-19 at 12 24 50 PM

**Comment by puqeko** [comment]
Has been deployed successfully

**Pull request merged**